### PR TITLE
Simplify path selection and remove unnecessary kernel registrations.

### PR DESCRIPTION
**Pull request by AdamHillier**
## What do these changes do?

This PR removes the overly-complicated Ruy path-search implementation and instead performs path selection with a single constexpr dependent on `AccumScalar` and `DstScalar`.  This makes path selection much cleaner, and also allows us to strip out the no-op kernels that previously had to be registered for every possible Ruy path (including all the various x86 AVX varieties).

## How Has This Been Tested?

CI.

## Benchmark Results

N/A.

## Related issue number

N/A.

### CODE REVIEW

**Review by Tombana — APPROVED** [review]
Nice find!

File: larq_compute_engine/core/bgemm_impl_ruy.h [hunk 1]
```diff
@@ -63,37 +63,27 @@ struct BGemmImplUsingRuy {
...
+#if RUY_PLATFORM_NEON
+    constexpr bool HasOptimizedNeonKernel =
+        std::is_same<AccumScalar, std::int16_t>::value ||
+        std::is_same<DstScalar, float>::value ||
+        std::is_same<DstScalar, std::int8_t>::value;
+    constexpr auto SelectedPath =
+        HasOptimizedNeonKernel ? ruy::Path::kNeon : ruy::Path::kStandardCpp;
+#else
+    constexpr auto SelectedPath = ruy::Path::kStandardCpp;
 #endif
```
Comment (AdamHillier) [inline larq_compute_engine/core/bgemm_impl_ruy.h:75]: This is the code that now does path selection.

File: larq_compute_engine/core/bgemm_kernels_arm64.h [hunk 2]
```diff
@@ -903,6 +903,9 @@ IF_BITPACKED_OUTPUT(
       "beq 60f\n"
 
       // Slow path: not all the 8x4 block fits
+      //     Fill v30.b[0] with a bitstring of the form 0...01...1 with `w1`
+      // ones, so that v30 can be used as a mask to keep only the bits in each
+      // bitpacked byte that correspond to actual output values.
```
Comment (AdamHillier) [inline larq_compute_engine/core/bgemm_kernels_arm64.h:908]: This was one of the review suggestions from #496 that I forgot to add before it got merged.

**Review by lgeiger — APPROVED** [review]
Great cleanup 👍

**Review by Tombana — APPROVED** [review]

**Pull request merged**